### PR TITLE
Stop caching the instance of HttpCallWrapper, which caused it to behave like a singleton class

### PR DESCRIPTION
**Pull request by RossAllenBell**
We discovered that the HttpCallWrapper instance was being shared between requests. These changes ensure it is used as a normal class/object, rather than what is effectively a singleton class. I expect this would resolve some thread-unsafe behavior.

These changes also remove the thread-unsafe behavior, but I think it's still a good idea to remove the singleton class behavior as well.

#### Pre-Review Checklist
- [X] Covered the changes with automated tests
- [X] Tested the changes locally
- [X] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

## Release Note

Fixed an issue where two requests to the `/documents` endpoint could interfere with each other in a thread-unsafe manner.

## For Elastic Internal Use Only
- [X] Considered corresponding documentation changes to [contribute separately](https://github.com/elastic/enterprise-search-pubs#contribute-docs-changes-for-product-changes)
- [X] New configuration settings added in this PR follow the [official guidelines](https://github.com/elastic/ent-search/blob/main/doc/enterprise-search-config.md)
- [X] [Manual test steps](https://github.com/elastic/connectors/blob/main/docs/INTERNAL.md#minimal-manual-tests)  are successful
- [ ] Enterprise Search builds successfully with a gem built from this PR's branch.
  - _Link the ent-search PR here_

### CODE REVIEW

**Comment by seanstory** [comment]
Disclaimer: I don't know anything about Sinatra.

I _think_ that this could be accomplished more cleanly with a "before filter". cmd+f that here: http://sinatrarb.com/intro.html

Screen Shot 2022-04-20 at 3 45 56 PM

**Review by seanstory — DISMISSED** [review]
LGTM, though see my note above about us potentially doing this with a `before` so as to not have to have `settings.connector_class.new` all over the app.rb

**Comment by wangch079** [comment]
Agree with @seanstory , the before filter seems to be a more clean solution.

File: lib/app/app.rb [hunk 1]
```diff
@@ -54,6 +54,8 @@ class ConnectorsWebApp < Sinatra::Base
   end
 
   before do
+    @connector = settings.connector_class.new
```
Comment (wangch079) [inline lib/app/app.rb:57]: I think with the change you made in `http_call_wrapper` (making `extractor` a method variable instead of instance variable), I think we can revert the changes here? Setting an instance of `http_call_wrapper` to `settings.connector` can avoid the instantiation in each request.

Comment (RossAllenBell) [reply lib/app/app.rb:57]: Yah, I ended up solving the problem twice, once in the _HttpCallWrapper_ implementation to make it thread-safe, and once in this controller code to make it instantiate a new _HttpCallWrapper_ on every request. I'd prefer to continue instantiating a new object on every request. The shared object approach was surprising and I would expect we'll accidentally introduce new regressions again in the future with it.

**Review by seanstory — APPROVED** [review]
LGTM, though do we need a test that would catch any thread safety regressions we might have? These are a pain to write and often cast a really small net, so probably not worth it, but thought I'd toss it out.

**Comment by RossAllenBell** [comment]
> LGTM, though do we need a test that would catch any thread safety regressions we might have? These are a pain to write and often cast a really small net, so probably not worth it, but thought I'd toss it out.

I looked into it for a few minutes earlier and came to the exact same conclusion.
